src/Altinn.App.Core/Features/Action/PaymentUserAction.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Altinn.App.Core.Features.Payment.Models;
+using Altinn.App.Core.Features.Payment.Services;
+using Altinn.App.Core.Internal.Data;
+using Altinn.App.Core.Internal.Process;
+using Altinn.App.Core.Internal.Process.Elements;
+using Altinn.App.Core.Internal.Process.Elements.AltinnExtensionProperties;
+using Altinn.App.Core.Models.UserAction;
+using Microsoft.Extensions.Logging;
+
+namespace Altinn.App.Core.Features.Action
+{
+    internal class PaymentUserAction : IUserAction
+    {
+        private readonly IProcessReader _processReader;
+        private readonly IDataService _dataService;
+        private readonly ILogger<PaymentUserAction> _logger;
+        private readonly IPaymentService _paymentService;
+
+        public PaymentUserAction(IProcessReader processReader, IDataService dataService, IPaymentService paymentService, ILogger<PaymentUserAction> logger)
+        {
+            _processReader = processReader;
+            _dataService = dataService;
+            _paymentService = paymentService;
+            _logger = logger;
+
+        }
+
+        public string Id => "pay";
+
+        public async Task<UserActionResult> HandleAction(UserActionContext context)
```

Comment:
This user action is invoked by the frontend whenever the user clicks "Pay/Betal" in the payment page. Basically just finds the redirect URL in storage and redirects the browser over to the payment provider.